README.md

```diff
@@ -36,19 +36,68 @@ scrape_configs:
     metrics_path: /export
     params:
       command: [check_load] # Run the check_load command.
+    relabel_configs:
+      - source_labels: [__address__]
+        target_label: __param_target
+      - source_labels: [__param_target]
+        target_label: instance
+      - target_label: __address__
+        replacement: 127.0.0.1:9275 # Nrpe exporter.
     static_configs:
       - targets: # Targets to run the specified command against.
         - '127.0.0.1:5666'
         - 'example.com:5666'
+
+  - job_name: nrpe_check_apt
+    honor_timestamps: true
+    params:
+      command: [check_apt]
+      ssl: [true]
+    scrape_interval: 3h # no need to poll this more frequently
```

Comment:
Scrape intervals over 2m are not safe with Prometheus, plus you should use the same scrape interval for everything to keep things manageable.

Reply:
My current server-side implementation of check_apt is pretty heavy weight: 
it does an 'apt-get update' and reports if any packages need to be updated. 
Clearly I don't want to execute this function every  2 minutes. 
 
In prometheus, I  mimicked the nagios long poll behavior and abused the alert by having it look back at the 
last report with this expression:
> expr: last_over_time(command_status{job="nrpe_check_apt"}[4h]) > 0
I understand that this not the way scraping is intended to be run, but thought 
this approach would have the lowest overall cost and allow me to keep the policy on the (prometheus) 
polling side.

An alternative approach would be to implement the 3h policy  on the NRPE server
by caching the results until the NRPE server deemed it was time to run the full script. 
One disadvantage with this approach is one cannot easily trigger a full execution from the 
NRPE client.  Another disadvantage is the excess network traffic.  For this particular case, 
neither disadvantage is a big deal ( however, I do need to rewrite my NRPE servers, which 
was the main driver for using this module.)
 
[ apologies for the delay - I did not get notification of your comments.   Will provide further comments tomorrow. ]

Reply:
Examples should not be encouraging non-recommended and overly complex practices like this. Please remove this example.

I'd also suggest you look at the node exporter textfile collector, it's better suited to cronjob monitoring.